cpp/include/cuspatial/experimental/geometry/segment.cuh

```diff
@@ -16,15 +16,26 @@
 
 #pragma once
 
+#include <cuspatial/cuda_utils.hpp>
 #include <cuspatial/vec_2d.hpp>
 
 namespace cuspatial {
 
 template <typename T>
-struct segment {
+class segment {
+ public:
   using value_type = T;
   vec_2d<T> first;
   vec_2d<T> second;
+
+  /// Return a copy of segment, translated by `v`.
+  segment<T> CUSPATIAL_HOST_DEVICE translate(vec_2d<T> const& v) const
+  {
+    return segment<T>{first + v, second + v};
+  }
+
+  /// Return the geometric center of segment.
+  vec_2d<T> CUSPATIAL_HOST_DEVICE center() const { return midpoint(first, second); }
```

Comment:
Would midpoint() be a better name?

Reply:
Aha! I thought about this too, and I think [`center`](https://en.wikipedia.org/wiki/Centre_(geometry))  is a more generic name for the geometric-center of the geometry. Think about deconditioning the floating point coordinates as inversely translating the object with the `-center` vector, do op and translate back. You can apply this sequence regardless of type of geometry you are dealing with. A box/circle may also have a geometry center and adopt the same sequence, albeit the implementation of `.center` is different. I imagine `.center` to exist in part of some super class `Geometry2D` that will cover everything from `point`, `segment`, `linestring_ref`, `polygon_ref` and expose `.center` as a virtual function.